src/pages/index.js

```diff
@@ -97,7 +152,71 @@ function handleAddCardFormSubmit(inputValues) {
 function handlePreviewPicture(data) {
   popupWithImage.openModal(data);
 }
+// Cards Delete Confirm
+const deleteConfirmPopup = new PopupWithConfirm({
+  popupSelector: "#popup-confirm-delete",
+});
+deleteConfirmPopup.setEventListeners();
+//this runs when you click the trash button on a card
+function handleDeleteClick(card, cardId) {
+  // here will be ID
+  // popupWithConfirm should be opened with ID
+
+  deleteConfirmPopup.openModal();
+  deleteConfirmPopup.setSubmitHandler(() => {
+    //this runs when you submit the delte-confirm modal
+    deleteConfirmPopup.renderLoading(true);
+
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        card.remove();
+        deleteConfirmPopup.closeModal();
+      })
+      .catch(console.err)
+      .finally(() => {
+        deleteConfirmPopup.renderLoading(false);
+      });
+  });
+}
+
+function handleLikeButtonClick(card) {
+  console.log(card.isLiked);
+
+  //don't forget to add catch blocks
 
+  //check if the card is liked or not
+  if (card.isLiked) {
+    //if it is liked, we unlike it on the server and then unlike it on the Dom, and change isLiked to false
+    api
+      .disLikeCard(card.getId())
+      .then(() => {
+        card.unLikeCardOnDom();
+        this.isLiked = false;

```

Comment:
`this` should **not** be used **outside** classes in functions in `index.js` . This **context** is easy to lose. You need to find other ways of passing data.